dadi/lib/controller/clients.js

```diff
@@ -192,34 +202,34 @@ Clients.prototype.postResource = function (req, res, next) {
 
   // To add a resource to a client, the requesting client needs to have
   // "update" access to the "clients" resource, as well as access to the
-  // resource in question themselves.
+  // resource in question.
   return acl.access.get(req.dadiApiClient, 'clients').then(access => {
     if (access.update !== true) {
       return help.sendBackJSON(null, res, next)(
         acl.createError(req.dadiApiClient)
       )
     }
 
-    return acl.access.get(req.dadiApiClient, req.body.name)
-  }).then(access => {
-    let forbiddenType = Object.keys(req.body.access).find(type => {
-      return access[type] !== true
+    return acl.access.get(req.dadiApiClient, req.body.name).then(access => {
+      let forbiddenType = Object.keys(req.body.access).find(type => {
+        return access[type] !== true
+      })
+
+      if (forbiddenType) {
+        return help.sendBackJSON(null, res, next)(
+          acl.createError(req.dadiApiClient)
+        )
+      }
+
+      return model.resourceAdd(
+        req.params.clientId,
+        req.body.name,
+        req.body.access
+      ).then(({results}) => {
+        help.sendBackJSON(201, res, next)(null, {results})
+      }).catch(this.handleError(res, next))
     })
-
-    if (forbiddenType) {
-      return help.sendBackJSON(null, res, next)(
-        acl.createError(req.dadiApiClient)
-      )
-    }
-
-    return model.resourceAdd(
-      req.params.clientId,
-      req.body.name,
-      req.body.access
-    )
-  }).then(({results}) => {
-    help.sendBackJSON(201, res, next)(null, {results})
-  }).catch(this.handleError(res, next))
+  })
 }
 
 Clients.prototype.postRole = function (req, res, next) {
```

Comment:
Why not "createRole" over "postRole"?

Reply:
I've followed the existing nomenclature: controller says `post`, model says `create`. But happy to change it. 🙃